public/app/percona/shared/core/reducers/dbaas/addDBCluster/addDBCluster.ts

```diff
@@ -101,7 +104,7 @@ export const addDbClusterAction = createAsyncThunk(
         databaseImage: databaseVersion?.value,
         expose,
         internetFacing,
-        sourceRanges: sourceRanges?.map((item: any) => item?.sourceRange || ''),
+        ...(preparedSourceRanges.length > 0 && { sourceRanges: preparedSourceRanges }),
```

Comment:
What happens here if `preparedSourceRanges` is empty? If I'm not mistaken it would evaluate to `false` and the spread operator would cause a crash.

Reply:
Thank you very much for the question! I double-checked.

even if the field is empty

![image](https://user-images.githubusercontent.com/90199600/223669687-5e8f1211-2817-4251-bf95-a1287ebfc518.png)

the form always returns a value for it
![image](https://user-images.githubusercontent.com/90199600/223670042-3a29bae2-31c2-42ee-9d7e-5f0fb19ef7d7.png)

because if there is no value in the form, the `react-final-form-arrays` library refuses to display the field at all for `FieldArray` component.

So we have 2 different situations. 
1. Result of our reduce will be array with a values
2. Result of our reduce will be array with length=0

In case when reduce returns empty array we will get `false` in this condition
`...(preparedSourceRanges?.length > 0 && { sourceRanges: preparedSourceRanges }),`
but in our common `toAPI` function for PXC clusters 
![image](https://user-images.githubusercontent.com/90199600/223672017-a029cc6a-d907-4da0-945a-077c8abd1de2.png)

we are checking it => we will not write anything extra in the request

perhaps it could be done in some other way, but it seems to me that this is still a good way to cast to the DBCluster type

Reply:
Hmm, the issue is browser specific in chrome its fine:
![image](https://user-images.githubusercontent.com/32638572/223678143-69cfcd4e-2bb3-4f15-82d0-310f9266cd20.png)
but in firefox it throws an exception:
<img width="377" alt="image" src="https://user-images.githubusercontent.com/32638572/223677713-10524c8b-350b-4567-a32f-10c5229076f6.png">.
but when you assign it it's fine:
![image](https://user-images.githubusercontent.com/32638572/223678573-90adc69f-39ac-4a86-b497-555b1c7c1069.png)
If you have other properties it's fine also:
![image](https://user-images.githubusercontent.com/32638572/223678788-232dadd5-f67b-44fb-b3db-5748df1a697b.png)

I think it's fine in this case, but JS is WEIRD.